cluster-up/cluster/kind/manifests/kube-flannel.yaml

```diff
@@ -0,0 +1,531 @@
+spec:
+  privileged: false
+  volumes:
+    - configMap
+    - secret
+    - emptyDir
+    - hostPath
+  allowedHostPaths:
+    - pathPrefix: "/etc/cni/net.d"
+    - pathPrefix: "/etc/kube-flannel"
+    - pathPrefix: "/run/flannel"
+  readOnlyRootFilesystem: false
+  # Users and groups
+  runAsUser:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  fsGroup:
+    rule: RunAsAny
+  # Privilege Escalation
+  allowPrivilegeEscalation: false
+  defaultAllowPrivilegeEscalation: false
+  # Capabilities
+  allowedCapabilities: ['NET_ADMIN']
+  defaultAddCapabilities: []
+  requiredDropCapabilities: []
+  # Host namespaces
+  hostPID: false
+  hostIPC: false
+  hostNetwork: true
+  hostPorts:
+  - min: 0
+    max: 65535
+  # SELinux
+  seLinux:
+    # SELinux is unsed in CaaSP
+    rule: 'RunAsAny'
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: flannel
+rules:
+  - apiGroups: ['extensions']
+    resources: ['podsecuritypolicies']
+    verbs: ['use']
+    resourceNames: ['psp.flannel.unprivileged']
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - get
+  - apiGroups:
+      - ""
+    resources:
+      - nodes
+    verbs:
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - nodes/status
+    verbs:
+      - patch
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: flannel
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: flannel
+subjects:
+- kind: ServiceAccount
+  name: flannel
+  namespace: kube-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: flannel
+  namespace: kube-system
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: kube-flannel-cfg
+  namespace: kube-system
+  labels:
+    tier: node
+    app: flannel
+data:
+  cni-conf.json: |
+    {
+      "name": "cbr0",
+      "plugins": [
+        {
+          "type": "flannel",
+          "delegate": {
+            "hairpinMode": true,
+            "isDefaultGateway": true
+          }
+        },
+        {
+          "type": "portmap",
+          "capabilities": {
+            "portMappings": true
+          }
+        }
+      ]
+    }
+  net-conf.json: |
+    {
+      "Network": "10.244.0.0/16",
+      "Backend": {
+        "Type": "vxlan"
+      }
+    }
+---
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: kube-flannel-ds-amd64
+  namespace: kube-system
+  labels:
+    tier: node
+    app: flannel
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: flannel
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+      - operator: Exists
+        effect: NoSchedule
+      serviceAccountName: flannel
+      initContainers:
+      - name: install-cni
+        image: quay.io/coreos/flannel:v0.11.0-amd64
```

Comment:
the provider is failing to come online for me because of this image. 

```
Events:
  Type     Reason     Age                    From                                Message
  ----     ------     ----                   ----                                -------
  Normal   Scheduled  8m5s                   default-scheduler                   Successfully assigned kube-system/kube-flannel-ds-amd64-4skv4 to kind-1.14.2-control-plane
  Normal   Pulling    6m36s (x4 over 8m4s)   kubelet, kind-1.14.2-control-plane  Pulling image "quay.io/coreos/flannel:v0.11.0-amd64"
  Warning  Failed     6m36s (x4 over 8m4s)   kubelet, kind-1.14.2-control-plane  Failed to pull image "quay.io/coreos/flannel:v0.11.0-amd64": rpc error: code = Unknown desc = failed to resolve image "quay.io/coreos/flannel:v0.11.0-amd64": no available registry endpoint: failed to do request: Head https://quay.io/v2/coreos/flannel/manifests/v0.11.0-amd64: EOF
  Warning  Failed     6m36s (x4 over 8m4s)   kubelet, kind-1.14.2-control-plane  Error: ErrImagePull
  Warning  Failed     6m24s (x6 over 8m3s)   kubelet, kind-1.14.2-control-plane  Error: ImagePullBackOff
  Normal   BackOff    2m56s (x21 over 8m3s)  kubelet, kind-1.14.2-control-plane  Back-off pulling image "quay.io/coreos/flannel:v0.11.0-amd64"
```

Reply:
This is odd. Can you try just to pull the image locally? 
It looks like it's there https://quay.io/repository/coreos/flannel?tab=tags . Also, if I understood correctly @slintes did not have this kind of problem, so could it have been a temporary quay problem?

Reply:
yep, all good over here, just verified

`2019-07-19T12:22:55Z - kube-system - Normal - Pod kube-flannel-ds-amd64-xprn2 Pulled - Successfully pulled image "quay.io/coreos/flannel:v0.11.0-amd64"`